acme/acme/crypto_util.py

```diff
@@ -373,24 +376,50 @@ def get_names_from_subject_and_extensions(
 
 def _pyopenssl_cert_or_req_all_names(loaded_cert_or_req: Union[crypto.X509, crypto.X509Req]
                                      ) -> List[str]:
+    """
+    Deprecated
+    .. deprecated: 3.x.x
+    """
+    warnings.warn(
+        "_pyopenssl_cert_or_req_all_names is deprecated and will be removed in the "
+        "next major release of Certbot. Please use "
+        "acme.crypto_util.get_names_from_subject_and_extensions instead.", DeprecationWarning,
+        stacklevel=2
+    )
     cert_or_req = loaded_cert_or_req.to_cryptography()
     return get_names_from_subject_and_extensions(
         cert_or_req.subject, cert_or_req.extensions
     )
 
 
-def _pyopenssl_cert_or_req_san(cert_or_req: Union[crypto.X509, crypto.X509Req]) -> List[str]:
+def _cryptography_cert_or_req_san(
+    cert_or_req: Union[x509.Certificate, x509.CertificateSigningRequest],
+) -> List[str]:
     """Get Subject Alternative Names from certificate or CSR using pyOpenSSL.
 
     .. note:: Although this is `acme` internal API, it is used by
         `letsencrypt`.
 
     :param cert_or_req: Certificate or CSR.
-    :type cert_or_req: `OpenSSL.crypto.X509` or `OpenSSL.crypto.X509Req`.
+    :type cert_or_req: `x509.Certificate` or `x509.CertificateSigningRequest`.
 
     :returns: A list of Subject Alternative Names that is DNS.
     :rtype: `list` of `str`
 
+    """
+    # ???: is this translation needed?
```

Comment:
as opposed to just doing `san_ext = cert_or_req.extensions.get_extension_for_class()` below?